### PR TITLE
docs: add docs for openapi

### PR DESCRIPTION
**Pull request by ecmadao**
图片

图片

图片

### CODE REVIEW

File: content/docs/en/features/openapi/overview.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+title: Bytebase OpenAPI
+---
+
+Bytebase provides OpenAPI to allow users operating resources without the UX.
+
+It currently supports following OpenAPIs:
+
+- [SQL advise](/docs/features/openapi/sql-advise)
+
+We provide Swagger as OpenAPI online docs. Once you started your bytebase service, you can access it on `http://localhost:8080/swagger/index.html`.
```
Comment (dragonly) [inline content/docs/en/features/openapi/overview.md:11]:
```suggestion
Bytebase provides OpenAPI to allow users to operate resources without the UX.

It currently supports the following OpenAPIs:

- [SQL advise](/docs/features/openapi/sql-advise)

We provide Swagger as OpenAPI online docs. Once you have started your Bytebase service, you can access it on `http://localhost:8080/swagger/index.html`.
```

File: content/docs/en/features/openapi/sql-advise.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+---
+title: SQL Advise API
+---
+
+The SQL advise API provides SQL check based on your schema review policy.
```
Comment (tianzhou) [inline content/docs/en/features/openapi/sql-advise.md:5]:
```suggestion
The SQL Advise API provides SQL checks based on your schema review policy.
```

File: content/docs/en/features/openapi/sql-advise.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+> Before you start, you should configure the schema review policy on a specific environment. Please check [Schema Review](/docs/use-bytebase/schema-review/overview) for more information.
+
+### Endpoint
+
+```bash
+GET http://localhost:8080/v1/sql/advise
```
Comment (tianzhou) [inline content/docs/en/features/openapi/sql-advise.md:12]: Need to include curl since this is a bash

Comment (tianzhou) [reply content/docs/en/features/openapi/sql-advise.md:12]: also let's give a full example including the query parameters

File: content/docs/en/features/openapi/sql-advise.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+### Parameters (query)
```
Comment (tianzhou) [inline content/docs/en/features/openapi/sql-advise.md:15]:
```suggestion
### Query parameters
```

File: content/docs/en/features/openapi/sql-advise.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+#### environment
```
Comment (tianzhou) [inline content/docs/en/features/openapi/sql-advise.md:17]: Need to mark whether it's required or optional

File: content/docs/en/features/openapi/sql-advise.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  {
+    "code": "number",
+    "content": "string",
+    "status": "string",
+    "title": "string"
+  }
+]
+```
+
+- `code`: The error code. Should be `0` if no errors.
```
Comment (tianzhou) [inline content/docs/en/features/openapi/sql-advise.md:44]: Referring to https://www.bytebase.com/docs/reference/error-code ?

File: content/docs/en/features/openapi/sql-advise.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+```json
+{
+  "message": "string"
+}
+```
+
+### Example
+
+```bash
+curl http://localhost:8080/v1/sql/advise\?environment\=Dev\&statement\=SELECT%20\*%20FROM%20\`table\`\&databaseType\=MySQL
```
Comment (tianzhou) [inline content/docs/en/features/openapi/sql-advise.md:108]: The escape is annoying. Maybe enclose with double quotes?

**Review by tianzhou — CHANGES_REQUESTED** [review]
BTW, we can take a look at https://stripe.com/docs/api, https://vercel.com/docs/rest-api as references.

File: content/docs/en/features/openapi/sql-advise.md (hunk 7 above)
Comment (ecmadao) [reply content/docs/en/features/openapi/sql-advise.md]: Update
图片

**Review by d-bytebase — APPROVED** [review]

File: content/docs/en/features/openapi/sql-advise.md (hunk 6 above)
Comment (ecmadao) [reply content/docs/en/features/openapi/sql-advise.md]: The error code there is out of date. Now we introduce the namespace concept, maybe we should update it in another PR?

File: content/docs/en/api/overview.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
+---
+title: Bytebase OpenAPI
+---
+
+Bytebase provides OpenAPI to allow users to operate resources without the UX.
+
+It currently supports the following OpenAPIs:
+
+- [SQL advise](/docs/api/sql-advise)
```
Comment (tianzhou) [inline content/docs/en/api/overview.md:9]:
```suggestion
- [SQL Advise](/docs/api/sql-advise)
```

File: content/docs/en/api/overview.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
+---
+title: Bytebase OpenAPI
+---
+
+Bytebase provides OpenAPI to allow users to operate resources without the UX.
+
+It currently supports the following OpenAPIs:
```
Comment (tianzhou) [inline content/docs/en/api/overview.md:7]:
```suggestion
Bytebase supports the following OpenAPIs:
```

File: content/docs/en/api/overview.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+- [SQL advise](/docs/api/sql-advise)
+
+We provide Swagger as OpenAPI online docs. Once you have started your Bytebase service, you can access it on `http://localhost:8080/swagger/index.html`.
```
Comment (tianzhou) [inline content/docs/en/api/overview.md:11]:
```suggestion
We provide Swagger OpenAPI docs. Once you have started your Bytebase service, you can access it on `http://localhost:8080/swagger/index.html`.
```

**Review by tianzhou — APPROVED** [review]
LGTM

File: content/docs/en/api/sql-advise.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+### Query parameters
+
+| Parameter    | Required?    | Description                                                            | Example                |
+| ------------ | ------------ | ---------------------------------------------------------------------- | ---------------------- |
+| environment  | **Required** | The environment name for your schema review policy. **Case sensitive** | Dev                    |
+| databaseType | **Required** | The database type. Available values : `MySQL`, `PostgreSQL`, `TiDB`.   | MySQL                  |
+| statement    | **Required** | The SQL statement.                                                     | SELECT \* FROM `table` |
+
```
Comment (candy2255) [inline content/docs/en/api/sql-advise.md:27]: In my opinion, it is better to mark "required" as below
CleanShot 2022-07-07 at 14 49 52@2x

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**